app/facades/weather_facade.rb

```diff
@@ -0,0 +1,16 @@
+class WeatherFacade
+  def self.daily_weather(params)
+    results = WeatherService.weather_search(params)
+    if results[:cod].nil?
+      get_weather(results)
+    else
+      results
+    end
+  end
```

Comment:
Looks like line 4 is checking for an error code perhaps? If so, I like this solution to create `WeatherDay` objects in the event that the response is good.